### PR TITLE
MP-4748 🏗️ OpenAPI Specification v3.0.0

### PR DESCRIPTION
**Pull request by M4tini**
- Updated all schema's to [OpenAPI Specification v3.0.0](https://spec.openapis.org/oas/v3.0.0).
- Ported OpenAPI related changes from the api-specification.
- Properly require `id`, `files` and `collo_number` on shipment responses.

TODO
- [x] Rebase on develop + fix all conflicts
- [x] Refactor all new functionality since January 2022

### CODE REVIEW

File: specification/info.json [hunk 1]
```diff
@@ -12,6 +11,7 @@
     "name": "pricing",
     "url": "https://www.myparcel.com/pricing"
   },
+  "version": "1.8.0",
```
Comment (M4tini) [inline specification/info.json:14]:
```suggestion
  "version": "2.0.0",
```

File: root.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{
+  "openapi": "3.0.0",
+  "info": {
+    "$ref": "./specification/info.json"
+  },
+  "servers": [
+    {
+      "url": "localhost"
+    }
```
[the remark this answers is not in the document]
Comment (M4tini) [reply root.json]: Good point, I replaced localhost with `https://carrier.myparcel.com`. It does not affect validation, it's only used in the UI.

**Comment by M4tini** [comment]
The "Undefined property" errors you encountered will be resolved by this PR https://github.com/MyParcelCOM/microservice-skeleton/pull/255

**Comment by yoan-myparcel** [comment]
@M4tini does this imply that we'll have to do skeleton update on all microservices?

**Comment by M4tini** [comment]
@yoan-myparcel No, only if we make them use the next major version of the carrier-specification. We might want to make sure if our release script avoids major version bumps.

**Comment by yoan-myparcel** [comment]
@M4tini yolo lol

**Review by yoan-myparcel — APPROVED** [review]

**Pull request merged**